From 03fe7249007b61e6b52eab7cbd33e11540c60592 Mon Sep 17 00:00:00 2001
From: Luke Craig <lacraig3@gmail.com>
Date: Thu, 1 Oct 2026 19:36:54 -0400
Subject: [PATCH 40/42] igloo: drop the igloo_should_block_mount hook

igloo_driver's block_mounts was its only consumer; igloo_driver removes it
in "igloo: remove ops and hooks Penguin never consumes". Restores
fs/namespace.c to upstream.
---
 drivers/igloobase/igloo_weaksyms.c | 14 --------------
 fs/namespace.c                     | 11 -----------
 2 files changed, 25 deletions(-)

diff --git a/drivers/igloobase/igloo_weaksyms.c b/drivers/igloobase/igloo_weaksyms.c
index 4dd4388c..7b4db82f 100644
--- a/drivers/igloobase/igloo_weaksyms.c
+++ b/drivers/igloobase/igloo_weaksyms.c
@@ -17,20 +17,6 @@ igloo_syscall_return_t igloo_syscall_return_hook = NULL;
 EXPORT_SYMBOL(igloo_syscall_enter_hook);
 EXPORT_SYMBOL(igloo_syscall_return_hook);
 
-// Function pointer for override
-bool (*igloo_should_block_mount_module)(struct path *path);
-bool igloo_should_block_mount(struct path *path);
-bool igloo_should_block_mount(struct path *path)
-{
-    if (igloo_should_block_mount_module) {
-        return igloo_should_block_mount_module(path);
-    } else {
-        // printk(KERN_INFO "igloo_should_block_mount: default implementation called\n");
-        return false;
-    }
-}
-EXPORT_SYMBOL(igloo_should_block_mount);
-
 void (*igloo_sock_release_module)(struct socket *sock);
 void igloo_sock_release(struct socket *sock)
 {
diff --git a/fs/namespace.c b/fs/namespace.c
index 5c4c3fdb..487ba30b 100644
--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -2721,11 +2721,6 @@ char *copy_mount_string(const void __user *data)
 	return data ? strndup_user(data, PAGE_SIZE) : NULL;
 }
 
-#ifdef CONFIG_IGLOO
-// forward declare igloo_should_block_mount
-bool igloo_should_block_mount(struct path *path);
-#endif
-
 /*
  * Flags is a 32-bit value that allows up to 31 non-fs dependent flags to
  * be given to the mount() call (ie: read-only, no-dev, no-suid etc).
@@ -2759,12 +2754,6 @@ long do_mount(const char *dev_name, const char __user *dir_name,
 	retval = user_path(dir_name, &path);
 	if (retval)
 		return retval;
-	
-#ifdef CONFIG_IGLOO
-	if (igloo_should_block_mount(&path)) {
-		return 0;
-	}
-#endif
 
 	retval = security_sb_mount(dev_name, &path,
 				   type_page, flags, data_page);
